File: android/src/main/kotlin/com/mytiki/sdk/capture/receipt/capacitor/req/ReqScan.kt
package com.mytiki.sdk.capture.receipt.capacitor.req

import com.getcapacitor.JSObject
import com.mytiki.sdk.capture.receipt.capacitor.Account
import com.mytiki.sdk.capture.receipt.capacitor.AccountCommon

/**
 * Represents a request to perform a scan operation.
 *
 * @param data The [JSObject] containing the scan request data.
 */
class ReqScan(data: JSObject) : Req(data.getString("requestId") ?: "") {
    /**
     * The type of scan to perform, as specified by [ScanTypeEnum].
     */
    val dayCutOff: Int

    /**
     * The optional [Account] associated with the scan request, or `null` if not provided.
     */
    val account: Account?

    init {
        dayCutOff = data.getInteger("dayCutOff", 7)!!
        account =
            if (!data.getString("username").isNullOrEmpty() && data.getString("id") != "") {
                Account(
                    AccountCommon.fromSource(data.getString("id") ?: ""),
                    data.getString("username") ?: "",
                    data.getString("password"),
                    data.getBool("isVerified")
                )
            } else {
                null
            }
    }
}

File: ios/Plugin/Req/ReqScan.swift
/*
 * Copyright (c) TIKI Inc.
 * MIT license. See LICENSE file in root directory.
 */


import Foundation
import Capacitor

public struct ReqScan {
    // The type of scan to perform, as specified by [ScanTypeEnum].
    let scanType: ScanTypeEnum?
    // The optional [Account] associated with the scan request, or `null` if not provided.
    let account: Account?
    
    init(data: CAPPluginCall) {
        guard let scanType = ScanTypeEnum.allCases.first(where: { value in value.rawValue == data.getString("scanType")}) else {
            data.reject("Invalid ScanType")
            self.account = nil
            self.scanType = nil
            return
            }
        self.scanType = scanType
        guard let source = data.getString("source") else {
            self.account = nil
            return
        }
        self.account = Account(accountType: AccountCommon.defaults[source]!, user: data.getString("user") ?? "", password: data.getString("password"), isVerified: data.getBool("isVerified"))
        
    }
}
